### DISCUSSION TITLE
How to change formula font

### DISCUSSION THREAD

**Discussion by QiangF** [kjambunathan/org-mode-ox-odt#201] [Q&A] [answered]
The university want all formula font to be "Times New Roman", I can manually change it in libreoffice for one formula. Is there a better option?
[equation.odt](https://github.com/kjambunathan/org-mode-ox-odt/files/9226669/equation.odt)

**Comment by QiangF** [comment]
There is a macro that does that https://wiki.documentfoundation.org/Macros/Math/001
I have create a function that copies a fixed setting to the new formula, but it misses setting for formula height and width, so the document is broken.

    (defun org-odt--copy-formula-file (info src-file target-dir)
      "Returns the internal name of the file"
      (let* ((target-settings (concat target-dir "settings.xml"))
             (target-content (concat target-dir "content.xml")))
        ;; Create a directory for holding formula file.  Also enter it in
        ;; to manifest.
        (make-directory (concat (plist-get info :odt-zip-dir) target-dir))
        (org-odt-create-manifest-file-entry
         info
         "application/vnd.oasis.opendocument.formula" target-dir "1.2")
        ;; Copy over the formula file from user directory to zip
        ;; directory.
        (message "Embedding %s as %s..." src-file target-content)
        (let ((ext (file-name-extension src-file)))
          (cond
            ;; Case 1: Mathml.
            ((member ext '("mathml" "mml"))
             (copy-file src-file (concat (plist-get info :odt-zip-dir) target-content) 'overwrite))
            ;; Case 2: OpenDocument formula.
            ((string= ext "odf")
             (org-odt--zip-extract src-file "content.xml"
                                   (concat (plist-get info :odt-zip-dir) target-dir)))
            (t (error "%s is not a formula file" src-file))))
        ;; Copy settings
        (copy-file "/home/q/Desktop/thesis/styles/settings.xml" (concat (plist-get info :odt-zip-dir) target-settings) 'overwrite)
        ;; Enter the formula file in to manifest.
        (org-odt-create-manifest-file-entry info "text/xml" target-content)
        target-content))

**Comment by kjambunathan** [comment]
Are you using starmath or latex.  Which mathml converter? Start with
starmath, and if it works move on to latex.   Give me a sample org file you
want
 to fix.

**Reply by QiangF** [reply]
I am using starmath, test snippet:

    #+label: eq:1
    \begin{equation*}
    [Ti]+[N]=TiN_{(s)}
    \end{equation*}

Turns out the settings.xml file is create by libreoffice in "org-odt-transform".

**Reply by QiangF** [reply]
The fonts are specified in the settings.xml

**Reply by QiangF** [reply]
[settings.xml.txt](https://github.com/kjambunathan/org-mode-ox-odt/files/9226830/settings.xml.txt)
The settings contain size that is different for each formula.

**Reply by QiangF** [reply]
Maybe change the macro UpdateAll in OrgModeUtilities.oxt

**Comment by QiangF** [comment, marked as answer]
I think I got the answer:
To change the default fonts follow these steps:

    Select Format → Fonts.
    Choose the seven fonts as described above.
    Click Default.

This will save your settings as the default fonts to be used in a new formula.

To change the base size of the formula, select Format → Font Size and enter the size you want. All other font sizes are relative to the base size. The default is 12 pt.

This will change the base size for the current formula only. To change the default base size (applied to new formulas only), you should click Default before you close the dialog. There is no way to easily change the base size of many formulae.

**Reply by QiangF** [reply]
https://wiki.openoffice.org/wiki/Documentation/FAQ/Formula/How_do_I_change_the_fonts_used_in_my_formula%3F

**Comment by kjambunathan** [comment]
Have you looked at the Libreoffice math guide.

Odt exporter doesnt write settings.xml file.  Try zipping with a custom
settings.xml.  You may have to change the manifest.xm too.


I will look at this issue tomorrow.

**Comment by kjambunathan** [comment]
I could verify that it is picking up the settngs from the current user profile.  If you need to work with multiple setting simultaneously you can pass on a different `env` variable in to the `soffice` command.  That is modify `org-odt-converter-processes`.   (`env` is like the Emacs' `custom.el`)

```
$ soffice --help | grep env
   -env:<VAR>[=<VALUE>] Set a bootstrap variable. For example: to set          
                       -env:UserInstallation=file:///tmp/test      
```

Remember to open the `odt` file created by the ODT exporter (which doesn't have `settings.xml`), and save it again in `LibreOffice` (which will create `settings.xml`)

Without `settings.xml` ... I believe the equations will display differently for different users.  For example, it will display with one font on one user, but with totally different set of fonts on other users.
